xla/service/gpu/gemm_rewriter.cc

```diff
@@ -174,15 +174,22 @@ bool IsSupportedF8Pattern(HloInstruction *instr, HloInstruction *&x,
     return ShapeUtil::SameElementType(instr->shape(),
                                       instr->operand(0)->shape());
   };
+  auto allgather_allowed = [](const HloInstruction *instr) -> bool {
+    return instr->GetModule()->config().num_partitions() <= 1 ||
+           instr->GetModule()->config().use_spmd_partitioning();
+  };
```

Comment:
@jurahul does this check look OK? Specifically, is it safe to change the dtype of all-gather if there is one partition, even if SMPD is not used? There can be multiple replicas in this case, right?

Reply:
Yeah, I think the check is not sufficient. If we are doing pure data parallelism from tensorflow, num_partitions() will be 1 and num_replicas() will be > 1, and this is the MPMD case. Can it just be use_spmd_partitioning(). Or will that miss some cases?

Reply:
>Can it just be use_spmd_partitioning()?
Yes. Since AllGather only appears under SPMD or MPMD case, it's better to limit the test instead.